### PR TITLE
fix: pass http headers to hls.js

### PR DESCRIPTION
**Pull request by TheComputerM**
Use the xhrSetup option in hls.js to set the http headers for media when on the web.

Fixes https://github.com/media-kit/media-kit/issues/1042

Things used:
- https://github.com/video-dev/hls.js/blob/master/docs/API.md#xhrsetup

### CODE REVIEW

**Comment by alexmercerind** [comment]
Does this work @TheComputerM? Also, if possible, can you add a unit-test?

**Comment by TheComputerM** [comment]
I had to comment out the unsupported error for web but otherwise it is functional
```dart
Media(
        'https://webhook.site/.../test.m3u8',
        httpHeaders: {
          'X-Test': 'the cool man',
        },
      ),
```
image

**Comment by TheComputerM** [comment]
Don't merge it now, I run a few more tests

**Comment by TheComputerM** [comment]
@alexmercerind It works properly with my server which hosting m3u8 content requiring an auth header.

File: media_kit/lib/src/models/media/media_web.dart [hunk 1]
```diff
@@ -91,9 +91,9 @@ class Media extends Playable {
         httpHeaders =
             httpHeaders ?? cache[normalizeURI(resource)]?.httpHeaders {
     // Ensure httpHeaders are not null or empty to prevent using unsupported HTTP headers on the web
-    if (httpHeaders != null && httpHeaders.isNotEmpty) {
-      throw UnsupportedError('HTTP headers are not supported on web');
-    }
+    // if (httpHeaders != null && httpHeaders.isNotEmpty) {
```
Comment (TheComputerM) [inline media_kit/lib/src/models/media/media_web.dart:94]: Any better way of checking if this is valid?

**Comment by TheComputerM** [comment]
@alexmercerind

**Review by alexmercerind — APPROVED** [review]
LGTM!

**Pull request merged**